newIDE/app/src/Export/GenericExporters/OnlineWebExport/OnlineGameLink.js

```diff
@@ -125,6 +127,80 @@ const OnlineGameLink = ({
     [build, getAuthorizationHeader, profile]
   );
 
+  const tryUpdateAuthors = React.useCallback(
+    async (i18n: I18nType) => {
+      if (!profile || !game || !build) return false;
+
+      const authorAcls = getAclsFromUserIds(project.getAuthorIds().toJSArray());
+
+      try {
+        await setGameUserAcls(
+          getAuthorizationHeader,
+          profile.id,
+          project.getProjectUuid(),
+          { author: authorAcls }
+        );
+      } catch (error) {
+        console.error(
+          'Unable to update the authors:',
+          error.response || error.message
+        );
+        showErrorBox({
+          message:
+            i18n._(t`Unable to update the authors of the project.`) +
+            ' ' +
+            i18n._(t`Verify your internet connection or try again later.`),
+          rawError: error,
+          errorId: 'author-update-error',
+        });
+        return false;
+      }
+
+      return true;
+    },
+    [build, game, getAuthorizationHeader, profile, project]
+  );
+
+  const tryUpdateSlug = React.useCallback(
+    async (partialGameChange: PartialGameChange, i18n: I18nType) => {
+      if (!profile || !game || !build) return false;
+
+      const { userSlug, gameSlug } = partialGameChange;
+
+      if (userSlug && gameSlug && userSlug === profile.username) {
+        try {
+          await setGameSlug(
+            getAuthorizationHeader,
+            profile.id,
+            game.id,
+            userSlug,
+            gameSlug
+          );
+          setSlug({ username: userSlug, gameSlug: gameSlug, createdAt: 0 });
+        } catch (error) {
+          console.error(
+            'Unable to update the game slug:',
+            error.response || error.message
+          );
+          showErrorBox({
+            message:
+              i18n._(
+                t`Unable to update the game slug. A slug must be 6 to 30 characters long and only contains letters, digits or dashes.`
+              ) +
+              ' ' +
+              i18n._(t`Verify your internet connection or try again later.`),
```

Comment:
It's the same as above, can you see an elegant way to deduplicate it? I'm not sure there's one, but maybe you can see something

Reply:
Hmm not sure there's one that doesn't complexify it more 😞